### DISCUSSION TITLE
Add default Idp to switch to  when domain not found within email domains from IdP

### DISCUSSION THREAD

**Discussion by egeorges92** [sventorben/keycloak-home-idp-discovery#396] [Ideas / New features]
The goal is to set a default IdP in case no IDPs respond to the user's email domain.
PR to follow

**Comment by egeorges92** [comment]
PR : https://github.com/sventorben/keycloak-home-idp-discovery/pull/398

**Reply by sventorben** [reply]
This has been discussed before and can be achieved with KC out of the box. See this thread for example: https://github.com/sventorben/keycloak-home-idp-discovery/issues/34#issuecomment-1049176120

Therefore, I will probably not implement and maintain it.

**Reply by sventorben** [reply]
Getting the discussion back to this thread.

From https://github.com/sventorben/keycloak-home-idp-discovery/issues/34#issuecomment-2230573930

> Thank you, Sven, for the tipr.
> I try before the Identity Provider Redirector. The issue with it, is that it doesn't sent the login_hint to the IdP, and it's an essential requirement for my use case.

That is a totally different thing. Probably I am not setting the login hint session attribute properly in that case. I am using the exact same logic as Keycloak does for the redirect. So I assume that in case no home IdP can be found, I am not setting the login_hint.

**Reply by antikalk** [reply]
Hi @sventorben 

I also stumbled upon the issue that the login_hint is not forwarded by the `identity-provider-redirector`. Having a look at the implementation on Keycloak side, it rather looks like redirect function is always called with loginHint being null: https://github.com/keycloak/keycloak/blob/40f39f3f0953b70f8f6d012d25cfb2980e9c1041/services/src/main/java/org/keycloak/authentication/authenticators/browser/IdentityProviderAuthenticator.java#L77-L79

So even if your extension would set the login hint on the session, I guess it would still not be forwarded by Keycloak.

Created an issue on Keycloak side: https://github.com/keycloak/keycloak/issues/36396
